src/transaction_builder.js

```diff
@@ -334,43 +293,56 @@ var canBuildTypes = {
   'pubkeyhash': true
 }
 
-function buildFromInputData (input, scriptType, parentType, redeemScript, allowIncomplete) {
+function buildFromInputData (input, scriptType, allowIncomplete) {
```

Comment:
This entire function could probably take more from `CombineSignatures`, https://github.com/bitcoin/bitcoin/blob/d612837814020ae832499d18e6ee5eb919a87907/src/script/sign.cpp#L309

Though we'd also need to modify `TransactionBuilder.sign`